src/lib.rs

```diff
@@ -21,9 +21,14 @@ pub enum rustls_result {
 pub struct rustls_client_config {
     _private: [u8; 0],
 }
+// This is intentionally not #[repr(C)]. We only ever pass this to C by
+// reference, and C is not allowed to know about the internals. Also, by
+// leaving this with the default repr, we cause cbindgen to generate only
+// a forward declaration of this struct, not a definition.
 #[allow(non_camel_case_types)]
 pub struct rustls_client_session {
-    _private: [u8; 0],
+    session: ClientSession,
+    error: String,
```

Comment:
This is so that, in success cases, we can return a pointer to an empty string rather than NULL. It would be incorrect to access the memory in that case, but it seemed nice to prevent a potential NULL dereference, at least.